fobi_custom/plugins/form_handlers/collect_data/fobi_form_handlers.py

```diff
@@ -85,4 +78,44 @@ def plugin_data_repr(self):
     return self.data.__dict__
 
 
+def get_element_info(self, form_element):
+    plugin_data = form_element.plugin_data
+    json_plugin_data = json.loads(plugin_data)
+
+    if 'text' in json_plugin_data.keys():
+        return None
+
+    else:
+        name = json_plugin_data['name']
+
+        element_info = {
+            'name': name,
+            'saved_data': self.form.cleaned_data[name],
+            'label': json_plugin_data['label'],
+            'type': form_element.plugin_uid,
+            'position': form_element.position,
+        }
+
+        return element_info
+
+
+def get_element_data(self, form_element):
+    element_info = get_element_info(self, form_element)
+    saved_data = element_info['saved_data']
+
+    return saved_data
+
+
+def get_saved_data(self, plugin_uid, default_value=''):
+    try:
+        form_element = FormElementEntry.objects.get(form_entry_id=self.form_id, plugin_uid=plugin_uid)
+        saved_data = get_element_data(self, form_element)
+        if (plugin_uid in ['time_start', 'time_stop']) and saved_data:
+            saved_data = datetime.combine(self.today, saved_data).replace(tzinfo=self.timezone)
+    except FormElementEntry.DoesNotExist:
+        saved_data = default_value
```

Comment:
You could scope this exception handling to the specific call that might raise an exception by using `try`/`except`/`else`:

```python
    try:
        form_element = FormElementEntry.objects.get(form_entry_id=self.form_id, plugin_uid=plugin_uid)
    except FormElementEntry.DoesNotExist:
        saved_data = default_value
    else:
        saved_data = get_element_data(self, form_element)
        if (plugin_uid in ['time_start', 'time_stop']) and saved_data:
            saved_data = datetime.combine(self.today, saved_data).replace(tzinfo=self.timezone)
```

Reply:
Nice 😎